File: java/UpdateCredentials.java
public class UpdateCredentials {
    public static void main(String[] args) throws Exception {
        String trunk = System.getenv("TRUNK_SID");
        String cid = System.getenv("CREDENTIAL_ID");
        if (trunk == null || cid == null || trunk.isEmpty() || cid.isEmpty()) {
            System.err.println("Error: TRUNK_SID and CREDENTIAL_ID required");
            System.exit(1);
        }
        String fn = System.getenv("SIP_CRED_FRIENDLY_NAME");
        if (fn == null || fn.isEmpty()) fn = "updated_label";
        String json = "{\"friendly_name\":\"" + fn.replace("\\", "\\\\").replace("\"", "\\\"") + "\"}";
        System.out.println("Updating credential...");
        _Client.put("/trunks/" + trunk + "/credentials/" + cid, json);
    }
}

File: python/update_credentials.py
#!/usr/bin/env python3
"""Update credential by id (path). Optional user_name, password, friendly_name in body."""
from _client import put
import os

trunk = os.environ.get("TRUNK_SID")
cid = os.environ.get("CREDENTIAL_ID")
if not trunk or not cid:
    print("Error: TRUNK_SID and CREDENTIAL_ID required")
    exit(1)

body = {}
if os.getenv("SIP_CRED_FRIENDLY_NAME"):
    body["friendly_name"] = os.getenv("SIP_CRED_FRIENDLY_NAME")
if os.getenv("SIP_CRED_USERNAME"):
    body["user_name"] = os.getenv("SIP_CRED_USERNAME")
if os.getenv("SIP_CRED_PASSWORD"):
    body["password"] = os.getenv("SIP_CRED_PASSWORD")
if not body:
    body["friendly_name"] = os.getenv("SIP_CRED_FRIENDLY_NAME", "updated_label")

print("Updating credential...")
put(f"/trunks/{trunk}/credentials/{cid}", body)
